clang/lib/Driver/Driver.cpp

```diff
@@ -4642,39 +4647,93 @@ class OffloadingActionBuilder final {
       }
     }
 
-    /// Initialize the GPU architecture list from arguments - this populates `GpuArchList` from
-    /// `--cuda-gpu-arch` flags. Only relevant if compiling to CUDA. Return true if any
-    /// initialization errors are found.
+    /// Initialize the GPU architecture list from arguments - this populates
+    /// `GpuArchList` from `--offload-arch` flags. Only relevant if compiling to
+    /// CUDA or AMDGCN. Return true if any initialization errors are found.
+    /// FIXME: SPIR-V AOT targets should also use `offload-arch` to better fit
+    /// in the standard model.
```

Comment:
I doubt that SPIR-V AOT targets' CLI could be conveniently switched to this usage, as the backend compiler CLIs are defined beyond the scope of the DPC++ implementation. The direction could be defined in a more "general" manner, though - along the lines of:
```suggestion
    /// FIXME: Consider providing a unified driver option for exact offload architecture
    /// so that command lines for SPIR-V AOT and CUDA/AMDGCN targets would
    /// use a common option, and the exact spelling like "offload-arch" is set within
    /// the driver.
```
(perhaps there's a more concise way to formulate this)

Reply:
Yes that's true, I just meant in terms of command line interface, what about:
```suggestion
    /// FIXME: Consider defining a standard option name for "offload-arch" to
    /// use in both the SPIR-V AOT and CUDA/AMDGCN targets.
```

Reply:
Your suggestion is much better, thank you!

Reply:
This is in relation to the clang driver flags not the called tool flags (this is the purpose of the driver to do the translation).

For SPIR-V targets, the driver uses a sub arch provided with the device triple to do the processing. While this is a workable solution for the SPIR toolchain, this by-passes the existing mechanism used by clang (`BoundArch` thing) thus reinvent the wheel. This makes non SPIR toolchains harder to integrate and more prone to breakage.

Reply:
Oooh I see, so what about:
```suggestion
    /// FIXME: "offload-arch" and the BoundArch mechanism should also be
    // used in the SYCLToolChain for SPIR-V AOT to get the offload
    // architecture instead of the Triple sub-arch it currently uses.
```
@Naghasan ? Does that make more sense?

Reply:
Yes, nitpicking a bit:

```suggestion
    /// FIXME: "offload-arch" and the BoundArch mechanism should also be
    // used in the SYCLToolChain for SPIR-V AOT to track the offload
    // architecture instead of the Triple sub-arch it currently uses.
```

Reply:
@AGindinson are you happy with this update to the comment?

Reply:
@npmiller Works for me!